### PR TITLE
Disable Debug Assert Popups in Unit Tests

### PR DESCRIPTION
**Pull request by andrew-lunarg**
We run unit tests on CI so cannot have interactive UI elements blocking them.

### CODE REVIEW

**Comment by lunarpapillo** [comment]
Please include @juan-lunarg in reviews that are primarily CMake.

This looks awesome to me, but also has quite a bit of redundancy; I suspect it can be made more compact and easier to use by defining and using a CMake macro.  I'm sure Juan will know the best practice.

**Comment by bradgrantham-lunarg** [comment]
> Please include @juan-lunarg in reviews that are primarily CMake.
> 
> This looks awesome to me, but also has quite a bit of redundancy; I suspect it can be made more compact and easier to use by defining and using a CMake macro. I'm sure Juan will know the best practice.

The CMake is probably not good - I just pounded on my keyboard until the first pass worked.  But since this fixes an issue that potentially hangs Windows CI, can we follow this PR with a PR that fixes the CMake style for this file and linker option in the tools and test targets?

File: framework/application/CMakeLists.txt [hunk 1]
```diff
@@ -78,8 +78,18 @@ common_build_directives(gfxrecon_application)
...
     add_executable(gfxrecon_application_test "")
     target_sources(gfxrecon_application_test PRIVATE
-            ${CMAKE_CURRENT_LIST_DIR}/test/main.cpp)
+            ${CMAKE_CURRENT_LIST_DIR}/test/main.cpp
+            ${CMAKE_CURRENT_LIST_DIR}/../../tools/platform_debug_helper.cpp)
     target_link_libraries(gfxrecon_application_test PRIVATE gfxrecon_application)
+    if (MSVC)
+        # Force inclusion of "gfxrecon_disable_popup_result" variable in linking.
+        # On 32-bit windows, MSVC prefixes symbols with "_" but on 64-bit windows it doesn't.
+        if(NOT CMAKE_CL_64)
```
Comment (juan-lunarg) [inline framework/application/CMakeLists.txt:87]: Use [CMAKE_SIZEOF_VOID_P](https://cmake.org/cmake/help/latest/variable/CMAKE_SIZEOF_VOID_P.html) to distinguish between 32/64.

`CMAKE_CL_64` is discouraged for new projects.

```suggestion
        if(CMAKE_SIZEOF_VOID_P EQUAL 4)
```

**Review by juan-lunarg — COMMENTED** [review]
Are you using `ctest` to run your tests?

Due to CTest's use of libuv to launch test processes, all system debug popup windows are always blocked.

File: framework/application/CMakeLists.txt (hunk 1 above)
Comment (bradgrantham-lunarg) [reply framework/application/CMakeLists.txt:87]: Nope, `catch2`, and we've been discussing today that popup windows are not blocked.

Comment (juan-lunarg) [reply framework/application/CMakeLists.txt:87]: But given switching to ctest would be a big change I'd say this is fine.

Comment (andrew-lunarg) [reply framework/application/CMakeLists.txt:87]: I'll make the `CMAKE_SIZEOF_VOID_P` change.

Comment (bradgrantham-lunarg) [reply framework/application/CMakeLists.txt:87]: I suggest leaving that for another PR.

Comment (andrew-lunarg) [reply framework/application/CMakeLists.txt]: Oh, I just did it.
Should I back it out?

**Review by bradgrantham-lunarg — APPROVED** [review]

**Pull request merged**